main.go

```diff
@@ -82,6 +82,13 @@ func main() {
 				clicommand.ArtifactShasumCommand,
 			},
 		},
+		{
+			Name:  "env",
+			Usage: "Something something environment variables",
+			Subcommands: []cli.Command{
+				clicommand.EnvDumpCommand,
+			},
+		},
```

Comment:
```suggestion
		{
			Name:  "env",
			Usage: "Something something environment variables",
			Subcommands: []cli.Command{
				clicommand.EnvDumpCommand,
			},
			Action: clicommand.EnvDumpAction,
		},
```

I think something like this (with appropriate changes in `clicommand/env_dump.go`) should work to allow `buildkite-agent env` and `buildkite-agent env dump` to be synonymous.

Not only does it preserve backward compatibility (up to the flag changes you made), but I actually think this is desirable behaviour in the long term.

Reply:
> but I actually think this is desirable behaviour in the long term

I wonder. I don't think dumping the process' current env is very useful outside the one weird use-case we have internally, which we hope to eliminate via control sockets. I can't imagine a user actually wanting to use `buildkite-agent env` to see their environment as JSON. I'd rather it show the usage and list of subcommands.